src/Microsoft.SqlTools.Authentication/Authenticator.cs

```diff
@@ -0,0 +1,242 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensions.Msal;
+
+namespace Microsoft.SqlTools.Authentication
+{
+    /// <summary>
+    /// Provides APIs to acquire access token using MSAL.NET v4 with provided <see cref="AuthenticationParams"/>.
+    /// </summary>
+    public class Authenticator
+    {
+        private const string ApplicationClientId = "a69788c6-1d43-44ed-9ca3-b83e194da255";
+        private const string ApplicationName = "azuredatastudio";
+        private const string AzureTokenFolder = "Azure Accounts";
+        private const string MSAL_CacheName = "azureTokenCacheMsal-azure_publicCloud";
+        private const string RedirectUri = "http://localhost";
+
+        private static ConcurrentDictionary<string, IPublicClientApplication> s_pcaMap
```

Comment:
we aren't generally using type/scoping variable name prefixes, such as `s_`, in this codebase.  Generally better to use similar style throughout the code if possible, IMO.